api/apis/integration/service_binding_test.go

```diff
@@ -0,0 +1,280 @@
+package integration_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	. "code.cloudfoundry.org/cf-k8s-controllers/api/apis"
+	"code.cloudfoundry.org/cf-k8s-controllers/api/repositories"
+	servicesv1alpha1 "code.cloudfoundry.org/cf-k8s-controllers/controllers/apis/services/v1alpha1"
+	workloadsv1alpha1 "code.cloudfoundry.org/cf-k8s-controllers/controllers/apis/workloads/v1alpha1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var _ = Describe("ServiceBinding Handler", func() {
+	BeforeEach(func() {
+		appRepo := repositories.NewAppRepo(k8sClient, clientFactory, nsPermissions)
+		serviceInstanceRepo := repositories.NewServiceInstanceRepo(clientFactory, nsPermissions)
+		serviceBindingRepo := repositories.NewServiceBindingRepo(clientFactory, nsPermissions)
+		decoderValidator, err := NewDefaultDecoderValidator()
+		Expect(err).NotTo(HaveOccurred())
+		// set up apiHandler, repos, etc.
+		apiHandler := NewServiceBindingHandler(
+			logf.Log.WithName("TestServiceBinding"),
+			*serverURL,
+			serviceBindingRepo,
+			appRepo,
+			serviceInstanceRepo,
+			decoderValidator,
+		)
+		apiHandler.RegisterRoutes(router)
+	})
+
+	Describe("POST /v3/service_credential_bindings", func() {
+		const (
+			appGUID             = "the-app-guid"
+			serviceInstanceGUID = "the-service-instance-guid"
+			secretName          = "the-secret-name"
+		)
+
+		var spaceGUID string
+
+		BeforeEach(func() {
+			org := createOrgAnchorAndNamespace(ctx, rootNamespace, generateGUID())
+			space := createSpaceAnchorAndNamespace(ctx, org.Name, "spacename-"+generateGUID())
+			spaceGUID = space.Name
+
+			role := createClusterRole(ctx, repositories.SpaceDeveloperClusterRoleRules)
+			createRoleBinding(ctx, userName, role.Name, spaceGUID)
+
+			cfApp := &workloadsv1alpha1.CFApp{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      appGUID,
+					Namespace: spaceGUID,
+				},
+				Spec: workloadsv1alpha1.CFAppSpec{
+					Name:         "name-for-" + appGUID,
+					DesiredState: "STOPPED",
+					Lifecycle: workloadsv1alpha1.Lifecycle{
+						Type: "buildpack",
+						Data: workloadsv1alpha1.LifecycleData{Buildpacks: []string{"java"}},
+					},
+				},
+			}
+			Expect(
+				k8sClient.Create(context.Background(), cfApp),
+			).To(Succeed())
+
+			_ = createServiceInstance(context.Background(), k8sClient, serviceInstanceGUID, spaceGUID, "service-instance-name", secretName)
+		})
+
+		When("type is `app`", func() {
```

Comment:
What other types are there? If `app` is the only option, this `When` is misleading and not needed